### PR TITLE
[UTOPIA-854] [Refactor] New Table Component. Replaces List for AdditionalRisks tab

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- Created new common component for Tabular Data - Table. 
- Replaces the existing List component usage inside Additional Risks tab
- Also includes super cool way of updating nested states 🤩

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?
Look & feel - good. #optimized

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [x] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Review by antsand — COMMENTED** [review]
Will some of this logic be changed in next sprint when we modify the table to be responsive?

File: src/frontend/src/components/common/Table/index.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      return;
+    }
+
+    setData((prevData) => prevData.filter((row, i) => i !== index));
+  };
+
+  // share updated data whenever there is a change
+  useEffect(() => {
+    if (onChangeHandler && !deepEqual(initialData, data)) {
+      console.log('inside', data);
```
Comment (binggaofw) [inline src/frontend/src/components/common/Table/index.tsx:65]: console.log can be removed

File: src/frontend/src/components/public/PIAFormTabs/additionalRisks/index.tsx [hunk 2]
```diff
@@ -43,98 +32,23 @@ const PIAAdditionalRisks = () => {
...
   };
 
-  const columns = [{ name: 'Possible risk' }, { name: 'Response' }];
+  const columns: Array<ColumnMetaData> = [
+    { key: 'risk', displayName: 'Possible risk', numberedLabelPrefix: 'Risk' },
+    { key: 'response', displayName: 'Response' },
+  ];
 
+  // passing updated data to parent for auto-save to work efficiently only if there are changes
   useEffect(() => {
```
Comment (binggaofw) [inline src/frontend/src/components/public/PIAFormTabs/additionalRisks/index.tsx:52]: duplicate useEffect call

File: src/frontend/src/components/common/Table/index.tsx (hunk 1 above)
Comment (kushal-arora-fw) [reply src/frontend/src/components/common/Table/index.tsx]: removed

File: src/frontend/src/components/public/PIAFormTabs/additionalRisks/index.tsx (hunk 2 above)
Comment (kushal-arora-fw) [reply src/frontend/src/components/public/PIAFormTabs/additionalRisks/index.tsx]: removed. Great catch!

**Review by binggaofw — APPROVED** [review]
loos good to me

**Review by Bottle7 — APPROVED** [review]
This is AWESOME! Thanks, Kushal! 🙌🏼 🚀

**Pull request merged**